src/geo/projection/globe_util.js

```diff
@@ -499,17 +499,40 @@ export function globeUseCustomAntiAliasing(painter: Painter, context: Context, t
     return transitionT === 0.0 && !useContextAA && !disabled && hasStandardDerivatives;
 }
 
-export function getGridMatrix(id: CanonicalTileID, bounds: LngLatBounds, latitudinalLod: number): Array<number> {
+export function getGridMatrix(id: CanonicalTileID, bounds: LngLatBounds, latitudinalLod: number, worldSize: number): Array<number> {
     const n = bounds.getNorth();
     const s = bounds.getSouth();
     const w = bounds.getWest();
     const e = bounds.getEast();
-    const S = 1.0 / GLOBE_VERTEX_GRID_SIZE;
-    const x = (e - w) * S;
-    const latitudinalSubdivs = GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
-    const y = (s - n) / latitudinalSubdivs;
-    const tileZoom = 1 << id.z;
-    return [0, x, tileZoom, y, 0, id.y, n, w, S];
+
+    // Construct transformation matrix for converting tile coordinates into LatLngs
+    const tiles = 1 << id.z;
+    const tileWidth = e - w;
+    const tileHeight = n - s;
+    const toUv = 1.0 / GLOBE_VERTEX_GRID_SIZE;
+    const x = tileWidth * toUv;
+    const y = -tileHeight / GLOBE_LATITUDINAL_GRID_LOD_TABLE[latitudinalLod];
+
+    const matrix = [0, x, 0, y, 0, 0, n, w, 0];
+
+    if (id.z > 0) {
+        // Add slight padding to patch seams between tiles.
+        const pixelPadding = 0.5;
+        const padding = pixelPadding * 360.0 / worldSize;
+
+        const xScale = padding / tileWidth + 1;
+        const yScale = padding / tileHeight + 1;
+        const padMatrix = [xScale, 0, 0, 0, yScale, 0, -0.5 * padding / x, 0.5 * padding / y, 1];
+
```

Comment:
The code here has now become more complicated, perhaps we could add a comment along the lines of:
```
// latLon = gridMatrix * (p_grid + scale * (p_grid - grid_center)) which implies
// latLon = grdiMatrix * (p_grid + scale * p_grid - scale * grid_center)
// and the part in parentheses is another transformation which can be represented by another matrix as follows:
// latLon = gridMatrix * padMatrix * p_grid
```

Reply:
This is fairly trivial transformation matrix composition. Perhaps I could add a comment or two about coordinate transformations

Reply:
indeed I was trying to capture the transformation from spaces as I think that might not be trivial at all for a person encountering it for the first time.